noir-projects/noir-protocol-circuits/crates/types/src/type_serialization.nr

```diff
@@ -78,3 +78,27 @@ impl Deserialize<FIELD_SERIALIZED_LEN> for Field {
         fields[0]
     }
 }
+
+impl <T, let N: u32, let M: u32> Serialize<N * M> for [T; N] where T: Serialize<M> {
+    fn serialize(self) -> [Field; N * M] {
+        let mut result: [Field; N * M] = std::mem::zeroed();
+        let mut serialized: [Field; M] = std::mem::zeroed();
+        let mut offset = 0;
+        for i in 0..N {
+            serialized = self[i].serialize();
+            for j in 0..M {
+                result[offset] = serialized[j];
```

Comment:
```suggestion
                result[i * M + j] = serialized[j];
```

My brain has apparently been damaged to the point where I find this easier to read :shrug: Not sure what the different compilation passes might end up resolving this to in both constrained and unconstrained contexts

Reply:
Ok I'll change it but it might generate bigger bytecode in public. We'll see.